### PR TITLE
bug 1420217. Update ES plugin that squashes stack on start

### PR DESCRIPTION
**Pull request by jcantrill**
blocked on https://github.com/fabric8io/openshift-elasticsearch-plugin/pull/77

### CODE REVIEW

**Comment by jcantrill** [comment]
[merge]  to consume release: https://fabric8-ci.fusesource.com/view/release%20builds/job/openshift-elasticsearch-plugin-release/43/

**Comment by jcantrill** [comment]
[test]

**Comment by richm** [comment]
```
Error from server (NotFound): templates.template.openshift.io "logging-fluentd-template" not found
```
Does the new test skip adding the fluentd template?  If so, we either need to add it back, or rewrite the tests to not use the fluentd template, but change the daemonset directly.

File: elasticsearch/run.sh [hunk 1]
```diff
@@ -161,7 +162,8 @@ seed_searchguard(){
         -tst JKS \
         -tspass tspass \
         -nhnv \
-        -icl ${DEBUG:+-dg}
+        -sniff true \
```
Comment (richm) [inline elasticsearch/run.sh]: what does -sniff do?

File: elasticsearch/run.sh [hunk 2]
```diff
@@ -161,7 +162,8 @@ seed_searchguard(){
         -tst JKS \
         -tspass tspass \
         -nhnv \
-        -icl ${DEBUG:+-dg}
```
Comment (richm) [inline elasticsearch/run.sh]: Why get rid of the ability to turn on debug output?

**Comment by jcantrill** [comment]
[test]

File: elasticsearch/run.sh (hunk 1 above)
Comment (jcantrill) [inline elasticsearch/run.sh]: ``` Sniffing can be used to detected available nodes by using the ES cluster state API```

https://github.com/floragunncom/search-guard-docs/blob/master/sgadmin.md

**Comment by jcantrill** [comment]
[test]

**Comment by jcantrill** [comment]
[merge]

**Comment by jcantrill** [comment]
[merge]

**Comment by richm** [comment]
[merge]

**Comment by richm** [comment]
[test]

**Comment by jcantrill** [comment]
@richm :
```
<richm> jcantril: https://github.com/openshift/origin-aggregated-logging/pull/421 causes many, many errors in the Elasticsearch logs
<richm> jcantril: 2017-06-09 22:21:14 INFO  SearchGuardSSLPlugin:84 - Search Guard 2 plugin not available
```
This error is from the sgadmin tool when we invoke it to seed the ACLs. I'm not exactly sure why the tool reports this but it doesn't mean anything; we are able to seed the ACL without it.  I was unable to figure out how to make it go away.
```
<richm> jcantril: this is probably the culprit: /opt/app-root/src/run.sh: line 166: -sniff: command not found
```
This is a switch for SGAdmin, but I will investigate.  It could be we don't require it.

**Comment by jcantrill** [comment]
@richm @stevekuznetsov   I have concerns about the CI on this job.  It does not have the correct version of the Openshift Elasticsearch plugin.  The latest merge failure points me to [1].  The job that faild is [2] and referenceds ES logs [3].  These logs identify a stack which is what the PR is supposed to remove and errors on a line in the constructor which does not match [4]

Something is NOT correct here.  Either the build is getting the wrong change, deploying the wrong image, or pulling the wrong logs.

[1] https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin_aggregated_logging/31/
[2] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_aggregated_logging_ansible/103/s3/
[3] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_aggregated_logging_ansible/103/s3/download/scripts/entrypoint/logs/containers/k8s_elasticsearch_logging-es-data-master-zfrcrfdg-1-1x99h_logging_49bc390c-4d7e-11e7-b559-0e64f3e96d1c_0.log
[4] https://github.com/fabric8io/openshift-elasticsearch-plugin/blob/3620a60b6ea1ad8deeb438a6b81b867ab8b745c9/src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

**Comment by jcantrill** [comment]
[merge] flake? curator test seemed fine locally

**Comment by jcantrill** [comment]
backport 1.5 in #519

**Pull request merged**